`make atdd` fails with 14 errors

Using commit #17dbef0129dc6cee447e743b7c16e1456e6e258c, on a macos (sierra, 10.12.6), running:

1. `make build-latest`
2. `make up-test-mode`
3. `make atdd`

The last step generates 14 errors.

The error message:

[ERROR] Tests run: 492, Failures: 14, Errors: 0, Skipped: 53, Time elapsed: 805.312 s <<< FAILURE! - in org.bitbucket.openkilda.atdd._RunCucumberTest
[ERROR] And topology contains 8 links(Scenario: Creation of Small Linear Network Topology)  Time elapsed: 0.006 s  <<< FAILURE!
java.lang.AssertionError: expected:<8> but was:<0>

[ERROR] Scenario: Creation of Small Linear Network Topology  Time elapsed: 0.006 s  <<< FAILURE!
java.lang.AssertionError: expected:<8> but was:<0>

[ERROR] Then traffic flows through ffr flow(Scenario: Port Failover followed by failure followed by recovery)  Time elapsed: 0.003 s  <<< FAILURE!
java.lang.AssertionError

[ERROR] Scenario: Port Failover followed by failure followed by recovery  Time elapsed: 0.004 s  <<< FAILURE!
java.lang.AssertionError

[ERROR] Then traffic flows through ffr flow(Scenario: ISL failover followed by failure followed by recovery)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError

[ERROR] Scenario: ISL failover followed by failure followed by recovery  Time elapsed: 0.002 s  <<< FAILURE!
java.lang.AssertionError

[ERROR] Then traffic flows through ffr flow(Scenario: Switch failover followed by failure followed by recovery)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError

[ERROR] Scenario: Switch failover followed by failure followed by recovery  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError

[ERROR] When flow pcet with 00:00:00:00:00:00:00:02 1 0 and 00:00:00:00:00:00:00:07 2 0 and 2000000 could be deleted(Scenario: Link Bandwidth Tracking)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: expected:<0> but was:<1>

[ERROR] Scenario: Link Bandwidth Tracking  Time elapsed: 0.002 s  <<< FAILURE!
java.lang.AssertionError: expected:<0> but was:<1>

[ERROR] Given a clean flow topology(Scenario: Link Bandwidth Limit)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: expected:<0> but was:<1>

[ERROR] Scenario: Link Bandwidth Limit  Time elapsed: 0.002 s  <<< FAILURE!
java.lang.AssertionError: expected:<0> but was:<1>

[ERROR] Given a clean flow topology(Scenario: Link Bandwidth No Enough Bandwidth)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: expected:<0> but was:<1>

[ERROR] Scenario: Link Bandwidth No Enough Bandwidth  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: expected:<0> but was:<1>

The detailed error area:

  @MVP1
  Scenario: Creation of Small Linear Network Topology # FlowCrudBasicInit.feature:8
    
    This scenario creates small linear network topology and makes sure topology is learned.

    Given a clean flow topology                       # Common.a_clean_flow_topology()
    And a clean controller                            # Common.a_clean_controller()
    And a nonrandom linear topology of 5 switches     # FlowCrudBasicRunTest.a_multi_path_topology()
    And topology contains 8 links                     # FlowPathTest.topologyContainsLinks(int)
      java.lang.AssertionError: expected:<8> but was:<0>
      	at org.junit.Assert.fail(Assert.java:88)
      	at org.junit.Assert.failNotEquals(Assert.java:834)
      	at org.junit.Assert.assertEquals(Assert.java:645)
      	at org.junit.Assert.assertEquals(Assert.java:631)
      	at org.bitbucket.openkilda.atdd.FlowPathTest.topologyContainsLinks(FlowPathTest.java:139)
      	at ✽.And topology contains 8 links(FlowCrudBasicInit.feature:15)
      

  @MVP1
  Scenario: Port Failover followed by failure followed by recovery # FlowFFR.feature:5
    
    This scenario checks that failover and recovery happens orderly and that
    failures do not break things apart.

    Given a clean flow topology                                    # Common.a_clean_flow_topology()
    And a clean controller                                         # Common.a_clean_controller()
    And a multi-path topology                                      # FlowPathTest.a_multi_path_topology()
    And topology contains 16 links                                 # FlowPathTest.topologyContainsLinks(int)
    And a flow ffr is successfully created                         # FlowFFRTest.successfulFlowCreation(String)
    And flow ffr in UP state                                       # NorthboundRunTest.flowState(String,String)
    And flow ffr path is shortest                                  # FlowFFRTest.flowPathIsShortest(String)
    And traffic flows through ffr flow                             # FlowFFRTest.trafficFlows(String)
    When a switch 00:00:00:00:00:00:00:03 port 1 is disabled       # FlowFFRTest.switchPortDisable(String,int)
    And flow ffr in UP state                                       # NorthboundRunTest.flowState(String,String)
    And flow ffr path is alternate                                 # FlowFFRTest.flowPathIsAlternate(String)
    Then traffic flows through ffr flow                            # FlowFFRTest.trafficFlows(String)
      java.lang.AssertionError
      	at org.junit.Assert.fail(Assert.java:86)
      	at org.junit.Assert.assertTrue(Assert.java:41)
      	at org.junit.Assert.assertTrue(Assert.java:52)
      	at org.bitbucket.openkilda.atdd.FlowFFRTest.trafficFlows(FlowFFRTest.java:76)
      	at ✽.Then traffic flows through ffr flow(FlowFFR.feature:22)
      
 
  @MVP1
  Scenario: ISL failover followed by failure followed by recovery   # FlowFFR.feature:35
    Given a clean flow topology                                     # Common.a_clean_flow_topology()
    And a clean controller                                          # Common.a_clean_controller()
    And a multi-path topology                                       # FlowPathTest.a_multi_path_topology()
    And topology contains 16 links                                  # FlowPathTest.topologyContainsLinks(int)
    And a flow ffr is successfully created                          # FlowFFRTest.successfulFlowCreation(String)
    And flow ffr in UP state                                        # NorthboundRunTest.flowState(String,String)
    And flow ffr path is shortest                                   # FlowFFRTest.flowPathIsShortest(String)
    And traffic flows through ffr flow                              # FlowFFRTest.trafficFlows(String)
    When an isl switch 00:00:00:00:00:00:00:03 port 1 is failed     # FlowFFRTest.anIslSwitchPortFails(String,int)
    And flow ffr in UP state                                        # NorthboundRunTest.flowState(String,String)
    And flow ffr path is alternate                                  # FlowFFRTest.flowPathIsAlternate(String)
    Then traffic flows through ffr flow                             # FlowFFRTest.trafficFlows(String)
      java.lang.AssertionError
      	at org.junit.Assert.fail(Assert.java:86)
      	at org.junit.Assert.assertTrue(Assert.java:41)
      	at org.junit.Assert.assertTrue(Assert.java:52)
      	at org.bitbucket.openkilda.atdd.FlowFFRTest.trafficFlows(FlowFFRTest.java:76)
      	at ✽.Then traffic flows through ffr flow(FlowFFR.feature:49)
      
 
  @MVP1
  Scenario: Switch failover followed by failure followed by recovery # FlowFFR.feature:62
    Given a clean flow topology                                      # Common.a_clean_flow_topology()
    And a clean controller                                           # Common.a_clean_controller()
    And a multi-path topology                                        # FlowPathTest.a_multi_path_topology()
    And topology contains 16 links                                   # FlowPathTest.topologyContainsLinks(int)
    And a flow ffr is successfully created                           # FlowFFRTest.successfulFlowCreation(String)
    And flow ffr in UP state                                         # NorthboundRunTest.flowState(String,String)
    And flow ffr path is shortest                                    # FlowFFRTest.flowPathIsShortest(String)
    And traffic flows through ffr flow                               # FlowFFRTest.trafficFlows(String)
    When a switch 00:00:00:00:00:00:00:03 is disconnected            # FlowFFRTest.aSwitchDisconnected(String)
    And flow ffr in UP state                                         # NorthboundRunTest.flowState(String,String)
    And flow ffr path is alternate                                   # FlowFFRTest.flowPathIsAlternate(String)
    Then traffic flows through ffr flow                              # FlowFFRTest.trafficFlows(String)
      java.lang.AssertionError
      	at org.junit.Assert.fail(Assert.java:86)
      	at org.junit.Assert.assertTrue(Assert.java:41)
      	at org.junit.Assert.assertTrue(Assert.java:52)
      	at org.bitbucket.openkilda.atdd.FlowFFRTest.trafficFlows(FlowFFRTest.java:76)
      	at ✽.Then traffic flows through ffr flow(FlowFFR.feature:76)
      
  @MVP1
  Scenario: Link Bandwidth Tracking                                                                                            # FlowPath.feature:6
    
    This scenario creates small multi-path network topology
    and checks that link available bandwidth is tracked.

    Given a clean flow topology                                                                                                # Common.a_clean_flow_topology()
    And a clean controller                                                                                                     # Common.a_clean_controller()
    And a clean flow topology                                                                                                  # Common.a_clean_flow_topology()
    And a multi-path topology                                                                                                  # FlowPathTest.a_multi_path_topology()
    And topology contains 16 links                                                                                             # FlowPathTest.topologyContainsLinks(int)
    Then all links have available bandwidth 9000000                                                                            # FlowPathTest.checkAvailableBandwidth(int)
    When flow pcet creation request with 00:00:00:00:00:00:00:02 1 0 and 00:00:00:00:00:00:00:07 2 0 and 1000000 is successful # FlowCrudBasicRunTest.successfulFlowCreation(String,String,int,int,String,int,int,int)
    And flow pcet with 00:00:00:00:00:00:00:02 1 0 and 00:00:00:00:00:00:00:07 2 0 and 1000000 could be created                # FlowCrudBasicRunTest.checkFlowCreation(String,String,int,int,String,int,int,int)
    Then shortest path links available bandwidth have available bandwidth 8000000                                              # FlowPathTest.checkShortestPathAvailableBandwidthDecreased(int)
    When flow pcet with 00:00:00:00:00:00:00:02 1 0 and 00:00:00:00:00:00:00:07 2 0 and 1000000 could be updated with 2000000  # FlowCrudBasicRunTest.checkFlowUpdate(String,String,int,int,String,int,int,int,int)
    Then shortest path links available bandwidth have available bandwidth 7000000                                              # FlowPathTest.checkShortestPathAvailableBandwidthDecreased(int)
    When flow pcet with 00:00:00:00:00:00:00:02 1 0 and 00:00:00:00:00:00:00:07 2 0 and 2000000 could be deleted               # FlowCrudBasicRunTest.checkFlowDeletion(String,String,int,int,String,int,int,int)
      java.lang.AssertionError: expected:<0> but was:<1>
      	at org.junit.Assert.fail(Assert.java:88)
      	at org.junit.Assert.failNotEquals(Assert.java:834)
      	at org.junit.Assert.assertEquals(Assert.java:645)
      	at org.junit.Assert.assertEquals(Assert.java:631)
      	at org.bitbucket.openkilda.atdd.FlowCrudBasicRunTest.checkFlowDeletion(FlowCrudBasicRunTest.java:144)
      	at ✽.When flow pcet with 00:00:00:00:00:00:00:02 1 0 and 00:00:00:00:00:00:00:07 2 0 and 2000000 could be deleted(FlowPath.feature:22)
      
    Then all links have available bandwidth 9000000                                                                            # FlowPathTest.checkAvailableBandwidth(int)

====> Error: Northbound Get Flow = MessageError{correlation_id=1507438108375, timestamp=1507438108448, error-type=Object was not found, error-message=Can not get flow: Flow pcet-1507437490513-1507437490513 not found, error-description=splitter_bolt}

  @MVP1
  Scenario: Link Bandwidth Limit                                                                                                # FlowPath.feature:27
    
    This scenario creates small multi-path network topology
    and checks that in case of no enough available bandwidth on the shortest path,
    longer path is chosen.

    Given a clean flow topology                                                                                                 # Common.a_clean_flow_topology()
      java.lang.AssertionError: expected:<0> but was:<1>
      	at org.junit.Assert.fail(Assert.java:88)
      	at org.junit.Assert.failNotEquals(Assert.java:834)
      	at org.junit.Assert.assertEquals(Assert.java:645)
      	at org.junit.Assert.assertEquals(Assert.java:631)
      	at org.bitbucket.openkilda.flow.FlowUtils.cleanupFlows(FlowUtils.java:453)
      	at org.bitbucket.openkilda.atdd.Common.a_clean_flow_topology(Common.java:45)
      	at ✽.Given a clean flow topology(FlowPath.feature:33)
      
 ====> Error: Northbound Get Flow = MessageError{correlation_id=1507438109042, timestamp=1507438109117, error-type=Object was not found, error-message=Can not get flow: Flow pcet-1507437490513-1507437490513 not found, error-description=splitter_bolt}

  @MVP1
  Scenario: Link Bandwidth No Enough Bandwidth                                                                                  # FlowPath.feature:48
    
    This scenario creates small multi-path network topology
    and checks that in case of no enough available bandwidth on the both shortest and alternative paths,
    flow could not be created.

    Given a clean flow topology                                                                                                 # Common.a_clean_flow_topology()
      java.lang.AssertionError: expected:<0> but was:<1>
      	at org.junit.Assert.fail(Assert.java:88)
      	at org.junit.Assert.failNotEquals(Assert.java:834)
      	at org.junit.Assert.assertEquals(Assert.java:645)
      	at org.junit.Assert.assertEquals(Assert.java:631)
      	at org.bitbucket.openkilda.flow.FlowUtils.cleanupFlows(FlowUtils.java:453)
      	at org.bitbucket.openkilda.atdd.Common.a_clean_flow_topology(Common.java:45)
      	at ✽.Given a clean flow topology(FlowPath.feature:54)
      
  @MVP1
  Scenario: Switch is Dropped                                            # TopologyEventsBasic.feature:33
    Given a clean controller                                             # Common.a_clean_controller()
    And a random linear topology of 5 switches                           # TopologyDiscoveryBasicTest.a_random_linear_topology_of(int)
    When the controller learns the topology                              # TopologyDiscoveryBasicTest.the_controller_learns_the_topology()
    And multiple links exist between all switches                        # TopologyEventsBasicTest.multiple_links_exist_between_all_switches()
      cucumber.api.PendingException: TODO: implement me
      	at org.bitbucket.openkilda.atdd.TopologyEventsBasicTest.multiple_links_exist_between_all_switches(TopologyEventsBasicTest.java:27)
      	at ✽.And multiple links exist between all switches(TopologyEventsBasic.feature:38)

Replace assert with assertTrue in 1d4c38dc5dbe42cbb21d2c845126ba26c5f1b54c
And now I move from `java.lang.AssertionError: expected:<8> but was:<0>` in `Scenario: Creation of Small Linear Network Topology` to `java.lang.AssertionError: expected:<8> but was:<23>`. 

Got a lot of duplicates from topology-engine-rest:
https://gist.github.com/nikitamarchenko/4d53a44889eed2a0f4248ac92395ad23

--- a/services/src/atdd/src/test/resources/features/mvp.1/FlowFFR.feature
+++ b/services/src/atdd/src/test/resources/features/mvp.1/FlowFFR.feature
@@ -26,6 +26,35 @@ Feature: Flow failover, failure and recovery
     And flow ffr path is alternate
     Then traffic does not flow through ffr flow
 
+    When a switch 00:00:00:00:00:00:00:05 port 1 is enabled
+    Then traffic flows through ffr flow
+
+  @MVP.FUTURE
+  Scenario: Port Failover followed by failure followed by recovery
+
+  This scenario checks that failover and recovery happens orderly and that
+  failures do not break things apart. Also it checks whether alternate route
+  is picked after complet failure.
+
+    Given a clean flow topology
+    And a clean controller
+    And a multi-path topology
+    And topology contains 16 links
+    And a flow ffr is successfully created
+    And flow ffr in UP state
+    And flow ffr path is shortest
+    And traffic flows through ffr flow
+
+    When a switch 00:00:00:00:00:00:00:03 port 1 is disabled
+    And flow ffr in UP state
+    And flow ffr path is alternate
+    Then traffic flows through ffr flow
+
+    When a switch 00:00:00:00:00:00:00:05 port 1 is disabled
+    And flow ffr in DOWN state
+    And flow ffr path is alternate
+    Then traffic does not flow through ffr flow
+
     When a switch 00:00:00:00:00:00:00:03 port 1 is enabled
     And flow ffr in UP state
     And flow ffr path is shortest

